plugins/ssr/ssrequality.ml

```diff
@@ -124,54 +124,58 @@ let pf_typecheck t =
   Proofview.Unsafe.tclEVARS sigma
   end
 
+let tclMATCH_GOAL env sigma c t_ok t_fail =
+  Proofview.Goal.enter begin fun gl ->
+  match unify_HO env sigma (Proofview.Goal.concl gl) c with
+  | sigma ->
+    (* FIXME: why do we drop the evarmap? *)
```

Comment:
Yes please, set the sigma.

Reply:
Done, but this is still not totally correct. We generate too many universe constraints for constants that will never be used in case of failure of matching. But well, this needs some serious design API-wise that is probably not fit for this kind of local changes.